File: 2765. Longest Alternating Subarray/Solution.cpp
class Solution {
public:
    int alternatingSubarray(vector<int>& nums) {
        int n = nums.size(), high = 0, left = 0, right = 1;
        while (right < n) {
            while (right < n && nums[right] != nums[right - 1] + 1) right++;
            if (right == n) break;
            left = right - 1;
            int x = nums[left], y = nums[right];
            bool alt = false;
            while (right < n) {
                if ((alt && nums[right] != x) || (!alt && nums[right] != y)) break;
                right++;
                alt = !alt;
            }
            high = max(high, right - left);
        }
        return (high > 0) ? high : -1;
    }
};

File: 2765. Longest Alternating Subarray/Solution.py
class Solution:
    def alternatingSubarray(self, nums: List[int]) -> int:
        n, high, left, right = len(nums), 0, 0, 1
        while right < n:
            while right < n and nums[right] != nums[right - 1] + 1:
                right += 1
            if right == n: break
            left = right - 1
            x, y = nums[left], nums[right]
            alt = False
            while right < n:
                if (alt and nums[right] != x) or (not alt and nums[right] != y): break
                right += 1
                alt = not alt
            high = max(high, right - left)
        return high if high > 0 else -1
